madprops/serializers.py

```diff
@@ -61,98 +92,164 @@ class Property(model):
 
     _options_class = PropertySerializerOptions
 
-    def field_to_native(self, obj, field_name):
-        if obj is None:
-            return None
+    def __init__(self, *args, **kwargs):
+        super(PropertySerializer, self).__init__(*args, **kwargs)
+        self.opts = self._options_class(self.Meta)
+        self.Meta.list_serializer_class = ListToDictSerializer
 
-        return self._to_representation(getattr(obj, field_name).all())
+    def to_representation(self, data):
+        if isinstance(data, dict):
+            return {data['name']: self._get_value(data)}
+        else:
+            return {data.name: self._get_value(
+                {'name': data.name, 'value': data.value})}
 
-    @cached_property
-    def data(self):
-        return self._to_representation(self.objects)
+    def _get_value(self, data):
+        if data['name'] in self.opts.json_props:
+            return json.loads(data['value'])
+        return data['value']
 
-    def _to_representation(self, objects):
-        return dict((obj.name, self._get_value(obj)) for obj in objects)
+    def save(self, property_data=None):
```

Comment:
since we're inside Property serializer, I think we could omit `property_` prefix here